tests/unit/test_utils.py

```diff
@@ -550,5 +551,35 @@ def test_dns_style_not_used_for_get_bucket_location(self):
         self.assertEqual(request.url, original_url)
 
 
+class TestInstanceCache(unittest.TestCase):
+    def test_cache_method(self):
+        cache = {}
+        class DummyClass(object):
+            def __init__(self):
+                self._instance_cache = cache
+
+            @instance_cache
+            def add(self, x, y):
+                return x + y
+
+            @instance_cache
+            def sub(self, x, y):
+                return x - y
+
+        adder = DummyClass()
+        self.assertEqual(adder.add(2, 1), 3)
+        # This one's from cache.
+        self.assertEqual(adder.add(2, 1), 3)
```

Comment:
There's a check at the end of this test that the cache has three items in it.  I can add the check after each assertEqual call to make it more explicit.  The one thing that I'd like to avoid is having to hardcode the cache key construction in this test, as ideally that should be able to change without affecting any unit tests.